--- WORKSPACE.xla	2025-04-27 16:44:36.328891468 +0000
+++ WORKSPACE	2025-04-28 00:32:10.718788297 +0000
@@ -1,5 +1,10 @@
 # buildifier: disable=load-on-top
-workspace(name = "xla")
+workspace(name = "devcontainer.xla")
+
+local_repository(
+    name = "xla",
+    path = "xla",
+)
 
 # Initialize the XLA repository and all dependencies.
 #
@@ -9,23 +14,23 @@
 # a macro from that repository in the same file.
 
 # Initialize hermetic Python
-load("//third_party/py:python_init_rules.bzl", "python_init_rules")
+load("@xla//third_party/py:python_init_rules.bzl", "python_init_rules")
 
 python_init_rules()
 
-load("//third_party/py:python_init_repositories.bzl", "python_init_repositories")
+load("@xla//third_party/py:python_init_repositories.bzl", "python_init_repositories")
 
 python_init_repositories(
     requirements = {
-        "3.11": "//:requirements_lock_3_11.txt",
+        "3.11": "@xla//:requirements_lock_3_11.txt",
     },
 )
 
-load("//third_party/py:python_init_toolchains.bzl", "python_init_toolchains")
+load("@xla//third_party/py:python_init_toolchains.bzl", "python_init_toolchains")
 
 python_init_toolchains()
 
-load("//third_party/py:python_init_pip.bzl", "python_init_pip")
+load("@xla//third_party/py:python_init_pip.bzl", "python_init_pip")
 
 python_init_pip()
 
@@ -33,28 +38,28 @@
 
 install_deps()
 
-load(":workspace4.bzl", "xla_workspace4")
+load("@xla//:workspace4.bzl", "xla_workspace4")
 
 xla_workspace4()
 
-load(":workspace3.bzl", "xla_workspace3")
+load("@xla//:workspace3.bzl", "xla_workspace3")
 
 xla_workspace3()
 
-load(":workspace2.bzl", "xla_workspace2")
+load("@xla//:workspace2.bzl", "xla_workspace2")
 
 xla_workspace2()
 
-load(":workspace1.bzl", "xla_workspace1")
+load("@xla//:workspace1.bzl", "xla_workspace1")
 
 xla_workspace1()
 
-load(":workspace0.bzl", "xla_workspace0")
+load("@xla//:workspace0.bzl", "xla_workspace0")
 
 xla_workspace0()
 
 load(
-    "//third_party/gpus/cuda/hermetic:cuda_json_init_repository.bzl",
+    "@xla//third_party/gpus/cuda/hermetic:cuda_json_init_repository.bzl",
     "cuda_json_init_repository",
 )
 
@@ -66,7 +71,7 @@
     "CUDNN_REDISTRIBUTIONS",
 )
 load(
-    "//third_party/gpus/cuda/hermetic:cuda_redist_init_repositories.bzl",
+    "@xla//third_party/gpus/cuda/hermetic:cuda_redist_init_repositories.bzl",
     "cuda_redist_init_repositories",
     "cudnn_redist_init_repository",
 )
@@ -80,21 +85,21 @@
 )
 
 load(
-    "//third_party/gpus/cuda/hermetic:cuda_configure.bzl",
+    "@xla//third_party/gpus/cuda/hermetic:cuda_configure.bzl",
     "cuda_configure",
 )
 
 cuda_configure(name = "local_config_cuda")
 
 load(
-    "//third_party/nccl/hermetic:nccl_redist_init_repository.bzl",
+    "@xla//third_party/nccl/hermetic:nccl_redist_init_repository.bzl",
     "nccl_redist_init_repository",
 )
 
 nccl_redist_init_repository()
 
 load(
-    "//third_party/nccl/hermetic:nccl_configure.bzl",
+    "@xla//third_party/nccl/hermetic:nccl_configure.bzl",
     "nccl_configure",
 )
 
